### DISCUSSION TITLE
求助：克隆的项目副本无法正常编译，进行到第二步biber处理引文时自动终止，请问如何解决？

### DISCUSSION THREAD

**Discussion by iLoveVenki** [mengchaoheng/SCUT_thesis#63] [Q&A] [answered]
## 1. 问题描述
您好，本机副本通过git clone此git项目获得，所使用的TeX Live版本为2022，工具为vscode，但是无法正常编译，每次进行到第二步的biber阶段，编译过程会自动终止。此外，还出现了如下的vscode报错：
I found no \citation commands BibTeX
I found no \bibdata command BibTeX
I found no \bibstyle command BibTeX

> `上述错误，目前暂未解决，该错误有何关键影响，由于无法完整地完成编译的四个阶段，其是否存在后续影响，暂不未知`。

## 2.  本机副本的编译配置
编译配置参考的是此开源项目README.md中的“快速使用”部分、settings_files/settings.json文件中的设置、以及其中推荐的知乎博客（https://zhuanlan.zhihu.com/p/166523064）。 之后，尝试了所推荐B站视频（https://hm.ru/Xs2YAh 华南理工大学硕博士论文LaTeX模板使用入门——基本环境配置篇），在问题排查过程中，查看了项目讨论区中的讨论6（https://github.com/mengchaoheng/SCUT_thesis/discussions/6） 和其他讨论帖子。

具体而言，全局settings.json文件中对latex-workshop的编译配置尝试了两种方案：1）知乎博客+settings_files/settings.json的融合配置；2）参考B站演示视频中操作，在vscode自动生成配置的基础上添加和修改。

## 3. 编译配置的截图说明
![图1 编译配置与错误_1](https://github.com/mengchaoheng/SCUT_thesis/assets/62459600/a0cc082b-c10b-4f78-b4a8-cab9d9f54b1d)

![图2 编译配置与错误_2](https://github.com/mengchaoheng/SCUT_thesis/assets/62459600/a4668002-51ce-4990-838f-7a85f7603b2f)

![图3 编译配置与错误_3](https://github.com/mengchaoheng/SCUT_thesis/assets/62459600/d4548621-9f87-4d33-a936-6db31f5e583f)

**Comment by mengchaoheng** [comment]
先给点思路，1.找身边能编译通过的人，实在不行到3号楼317找我。2.overleaf？

抱歉帮不到更多。因为这种错误通常和电脑环境有关，只能做排除法。

**Reply by iLoveVenki** [reply]
谢谢，我并非华南理工的，overleaf感觉不太方便，编译工程量存在限制，vscode比较习惯了。我按照您的思路先尝试一下，在我另外一台非主力机上测试一下是否可以正常编译成功。

**Reply by iLoveVenki** [reply]
尝试了在第二台非主力机上进行编译，xelatex->biber->xelatex*2和xelatexmk两种编译链均可以正常编译。对比两种编译环境，texlive的版本号均为Tex 3.141592653，vscode版本均为最新版，texstudio的版本不一致，非主力机上要低一点，但是编译配置应该和texstudio无关。采用相同的latex-workshop插件配置，在主力机上无法正常编译。两台电脑的texlive的安装都是很久之前了，好像都是按照网上教程推荐安装的，不清楚是不是当时安装的勾选项存在细微差异，导致其中一台无法正常编译，我卸载掉现在的Texlive，再重新安装一下Tex 2022或者2023版本试一下看看。

**Comment by mengchaoheng** [comment]
选择xmk编译？
![截屏2024-03-31 15 32 53](https://github.com/mengchaoheng/SCUT_thesis/assets/43166007/f5890edf-f7a5-479a-bb96-df31136b2082)

**Reply by iLoveVenki** [reply]
你指的是xelatexmk吧，这一个我刚才尝试了，虽然底部显示编译正常（打钩），但是点击生成pdf目录中“参考文献”时，错误跳转到了附录1部分，且缺少参考文献列表页。此外，选择xelatexmk编译之后，vscode中新增了三个语法错误提示：
I found no \citation commands BibTeX
I found no \bibdata command BibTeX
I found no \bibstyle command BibTeX

**Comment by mengchaoheng** [comment]
其实用vsvcode反而带来很多干扰，需要设置比较麻烦，灵活的东西带来不确定。texstutio虽然死板反而更专用性。想离线使用可以试试。

**Reply by iLoveVenki** [reply]
texstudio之前用过，但是日常编程开发较多，vscode中添加markdown等说明性开发记录很方便，且git等第三方插件比较丰富，所以就想试试用vscode+git编译和管理latex，同时利用markdown等管理开发记录。您的开源项目对我帮助很大，之后只做过单.tex项目的vscode编译。

**Comment by iLoveVenki** [comment, marked as answer]
@mengchaoheng 说明：该错误和texlive的版本无关，经尝试重新安装了texlive  2023版本，但在采用xelatex->biber->xelatex*2编译链编译项目时，仍然在第二步自动终止。

### 最终的解决方法：
#### 第1步：启用错误消息弹出
在用户全局的settings.json中，将下面编译出错时设置是否消息弹出设置中，将原始教程中推荐的false，改成true。修改后的配置项具体如下：
    "latex-workshop.message.error.show": true,
    "latex-workshop.message.warning.show": true,

#### 第2步：获取错误原因
在上述设置更改完成后，重新使用xelatex->biber->xelatex*2编译链编译，在进行到第二步的时候，弹出了错误信息：**Error unlinking file C:\WINDOWS\TEMP\MvpkHbhXPI using unlink0 at C:\WINDOWS\TEMP\par-56656e6b69\cache-96d297aa7563e182b1dd640f406e913a11204801\inc\lib/IPC/Run3.pm line 128.**

#### 第3步：排查错误原因
经询问chatGPT可知，上述编译错误的大致原因是biber在编译的时候，产生的中间文件被锁定，导致无法正常删除，具体细节我也没有深究。GPT给出的建议是：您可以尝试在指定的位置（C:\WINDOWS\TEMP\MvpkHbhXPI）手动删除文件，以确认是否存在权限或文件锁定问题。如果手动删除成功，则可能是程序运行时的问题。

#### 第4步：修复错误
 手动删除C:\WINDOWS\TEMP\中的对应文件夹，清楚缓存文件，然后继续执行xelatex->biber->xelatex*2编译链编译，最终成功解决错误。

**Comment by mengchaoheng** [comment]


**Reply by iLoveVenki** [reply]
是的，chatGPT对于排查未知错误还是挺有用，也很感谢你的帮助！
